File: backend/olympic_it_project/src/main/java/org/example/olympic_ot_project/dto/exam/option/UpdateQuestionOptionRequest.java
package org.example.olympic_ot_project.dto.exam.option;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import org.example.olympic_ot_project.enity.Question;
import org.example.olympic_ot_project.enity.QuestionOption;

@Data
public class UpdateQuestionOptionRequest {
    @NotNull(message = "Ký hiệu đáp án không được để trống")
    private Character label;

    @NotBlank(message = "Nội dung câu hỏi không được để trống")
    private String contentText;

    private String imageUrl;

    private Boolean isCorrect;
}

File: frontend/olympic_it_project/olympic_it_project/lib/dto/admin_manager/question/update_question_option_request.dart
class UpdateQuestionOptionRequest {
  final String label;
  final String contentText;
  final String? imageUrl;
  final bool isCorrect;

  UpdateQuestionOptionRequest({
    required this.label,
    required this.contentText,
    this.imageUrl,
    required this.isCorrect,
  });

  Map<String, dynamic> toJson() {
    return {
      "label": label,
      "contentText": contentText,
      "imageUrl": imageUrl,
      "isCorrect": isCorrect,
    };
  }
}